daft/expressions/expressions.py

```diff
@@ -387,6 +393,28 @@ def not_null(self) -> Expression:
         expr = self._expr.not_null()
         return Expression._from_pyexpr(expr)
 
+    def is_in(self, items: _List[Any] | Expression) -> Expression:
+        """Checks if values in the Expression are in the provided list
+
+        Example:
+            >>> # [1, 2, 3] -> [True, False, True]
+            >>> col("x").is_in([1, 3])
+
+        Returns:
+            Expression: Boolean Expression indicating whether values are in the provided list
+        """
+        if not (isinstance(items, Expression) or isinstance(items, list)):
+            raise TypeError(f"expected a python list or Daft Expression, got {type(items)}")
+
+        if isinstance(items, list):
```

Comment:
We should support all the types we see here https://github.com/Eventual-Inc/Daft/blob/5814922581b48eac48ad7941957e1e9187ce1ae4/daft/table/table.py#L151

You can probably factor the inner loop into a helper function that takes in an item and always returns a series.